terraform/modules/origin/main.tf

```diff
@@ -1,97 +1,161 @@
-    path = "/"
+  assume_role {
+    role_arn = var.assume_role_arn
   }
 }
 
-resource "aws_lb_listener_rule" "origin-frontend" {
-  listener_arn = aws_lb_listener.origin.arn
-  priority     = 100
+provider "aws" {
+  region = "us-east-1"
+  alias  = "use1"
 
-  action {
-    type             = "forward"
-    target_group_arn = aws_lb_target_group.origin-frontend.arn
+  assume_role {
+    role_arn = var.assume_role_arn
   }
+}
+
+resource "aws_cloudfront_origin_access_identity" "cloudfront_s3_access" {
+  comment = "${local.mode}-origin ${var.workspace} cloudfront accessing the Rails assets s3 bucket"
+}
+
+resource "random_password" "origin_alb_x_custom_header_secret" {
+  length  = 128
+  special = false
+}
+
+resource "aws_secretsmanager_secret" "origin_alb_x_custom_header_secret" {
+  name = "${local.mode}_origin_${var.workspace}_alb_x_custom_header_secret"
+}
+
+resource "aws_secretsmanager_secret_version" "origin_alb_x_custom_header_secret" {
+  secret_id     = aws_secretsmanager_secret.origin_alb_x_custom_header_secret.id
+  secret_string = random_password.origin_alb_x_custom_header_secret.result
+}
+
+resource "aws_cloudfront_distribution" "origin" {
+  origin {
+    domain_name = aws_lb.origin.dns_name
+    origin_id   = local.origin_alb_id
+
+    custom_origin_config {
+      http_port              = 80
+      https_port             = 443
+      origin_protocol_policy = "https-only"
+      origin_ssl_protocols   = ["TLSv1.2"]
+    }
 
-  condition {
-    path_pattern {
-      values = ["/assets/frontend/*"]
+    #TODO: frequent rototation for additional security,
+    #see: https://docs.aws.amazon.com/AmazonCloudFront/latest/DeveloperGuide/restrict-access-to-load-balancer.html
+    custom_header {
+      name  = "X-Cloudfront-Token"
+      value = aws_secretsmanager_secret_version.origin_alb_x_custom_header_secret.secret_string
     }
+
   }
-}
 
-resource "aws_lb_target_group" "origin-static" {
-  name        = "${local.mode}-origin-static-${var.workspace}"
-  port        = 80
-  protocol    = "HTTP"
-  vpc_id      = var.vpc_id
-  target_type = "ip"
+  origin {
+    domain_name = var.rails_assets_s3_regional_domain_name
+    origin_id   = local.origin_s3_id
 
-  health_check {
-    path = "/templates/core_layout.html.erb"
+    s3_origin_config {
+      origin_access_identity = aws_cloudfront_origin_access_identity.cloudfront_s3_access.cloudfront_access_identity_path
+    }
   }
-}
 
-resource "aws_lb_listener_rule" "origin-static" {
-  listener_arn = aws_lb_listener.origin.arn
-  priority     = 99
+  enabled         = true
+  is_ipv6_enabled = false
+  comment         = "${local.mode}-origin ${var.workspace} CDN in front of origin ALB and s3 rails assets bucket"
+  web_acl_id      = aws_wafv2_web_acl.origin_cloudfront_web_acl.arn
+
+  aliases = var.workspace == "ecs" && local.mode == "www" ? ["www.ecs.${var.publishing_service_domain}", "www-origin.${var.external_app_domain}"] : ["${local.mode}-origin.${var.external_app_domain}"]
```

Comment:
You could use `var.live` instead of `local.mode == "www"`. Would be nice to have a similar thing for `workspace`, so we can check `var.is_default_workspace && var.live`. (Related thought: `is_live` might be a bit more self explanatory than `live`)

I think you could also avoid the repetition of the external app domain one with something like:

```terraform
  aliases = compact([
    var.is_default_workspace && local.is_live ? "www.ecs.${var.publishing_service_domain}" : null,
    "${local.mode}-origin.${var.external_app_domain}"
  ])
```

Reply:
thanks, I can do that